src/Controls/src/Build.Tasks/CompiledConverters/ColumnDefinitionCollectionTypeConverter.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+using Microsoft.Maui.Controls.Build.Tasks;
+using Microsoft.Maui.Controls.Xaml;
+using Mono.Cecil.Cil;
+using static Mono.Cecil.Cil.Instruction;
+using static Mono.Cecil.Cil.OpCodes;
+
+namespace Microsoft.Maui.Controls.XamlC
+{
+	class ColumnDefinitionCollectionTypeConverter : ICompiledTypeConverter
+	{
+		public IEnumerable<Instruction> ConvertFromString(string value, ILContext context, BaseNode node)
+		{
+			var module = context.Body.Method.Module;
+
+			if (!string.IsNullOrWhiteSpace(value))
+			{
+				value = value.Trim();
+				var gridlengthconverter = new GridLengthTypeConverter();
+				var parts = value.Split(',');
+
+				yield return Create(Ldc_I4, parts.Length);
+				yield return Create(Newarr, module.ImportReference(("Microsoft.Maui.Controls", "Microsoft.Maui.Controls", "ColumnDefinition")));
+				for(var i =0;i< parts.Length;i++){
+				        yield return Create(Dup);
+					yield return Create(Ldc_I4, i);
+					foreach (var instruction in gridlengthconverter.ConvertFromString(parts[i], context, node))
+						yield return instruction;
+					yield return Create(Newobj, module.ImportCtorReference(
```

Comment:
Does calling `ImportCtorReference` in a loop affect performance at all? I assume we would normally do this outside the loop.

Reply:
that's fine, as 
1/ it's compiler time
2/ Import*Reference calls and return values are cached